pkg/metrics/metrics.go

```diff
@@ -194,6 +194,14 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	for _, node := range nodes.Items {
 		nodeFound := false
 		for _, address := range endpoints.Subsets[0].Addresses {
+			if address.TargetRef == nil {
+				// TargetRef not found
+				continue
+			}
+			if address.TargetRef.Kind != "Node" {
```

Comment:
nit: consider collapsing the two if statements 
```golang
if address.TargetRef == nil || address.TargetRef.Kind != "Node" {
    // TargetRef not found or invalid
    continue
}
```

Reply:
Addressed.